kuma/wiki/views/list.py

```diff
@@ -142,65 +143,73 @@ def revisions(request, document_slug, document_locale):
     List all the revisions of a given document.
     """
     locale = request.GET.get('locale', document_locale)
-    document = get_object_or_404(Document.objects
-                                         .select_related('current_revision'),
-                                 locale=locale,
-                                 slug=document_slug)
-    if document.current_revision is None:
-        raise Http404
 
-    def get_previous(revisions):
-        for current_revision in revisions:
-            for previous_revision in revisions:
-                # we filter out all revisions that are not approved
-                # as that's the way the get_previous method does it as well
-                # also let's skip comparing the same revisions
-                if (not previous_revision.is_approved or
-                        current_revision.pk == previous_revision.pk):
-                    continue
-                # we stick to the first revision that we find
-                if previous_revision.created < current_revision.created:
-                    current_revision.previous_revision = previous_revision
-                    break
-        return revisions
+    # Load document with only fields for history display
+    try:
+        document = (Document.objects
```

Comment:
Good idea. I switched to ``get_object_or_404`` with a longer queryset, and now ``Http404`` is no longer imported.